internal/controller/cluster_monitoring_controller.go

```diff
@@ -126,43 +126,14 @@ func (r *ClusterMonitoringReconciler) reconcile(ctx context.Context, cluster *cl
 	// Add finalizer first if not set to avoid the race condition between init and delete.
 	// Note: Finalizers in general can only be added when the deletionTimestamp is not set.
 	if !controllerutil.ContainsFinalizer(cluster, monitoring.MonitoringFinalizer) {
-		// We use a patch rather than an update to avoid conflicts when multiple controllers are adding their finalizer to the ClusterCR
-		// We use the patch from sigs.k8s.io/cluster-api/util/patch to handle the patching without conflicts
-		logger.Info("adding finalizer", "finalizer", monitoring.MonitoringFinalizer)
-		patchHelper, err := patch.NewHelper(cluster, r.Client)
-		if err != nil {
-			return ctrl.Result{}, errors.WithStack(err)
-		}
-		controllerutil.AddFinalizer(cluster, monitoring.MonitoringFinalizer)
-		if err := patchHelper.Patch(ctx, cluster); err != nil {
-			logger.Error(err, "failed to add finalizer", "finalizer", monitoring.MonitoringFinalizer)
-			return ctrl.Result{}, errors.WithStack(err)
-		}
-		logger.Info("added finalizer", "finalizer", monitoring.MonitoringFinalizer)
-		return ctrl.Result{}, nil
+		return r.addFinalizer(ctx, cluster)
```

Comment:
So whenever you add the finalizer you return?

Reply:
Like it was done before

Reply:
I don't want to change the current implementation so I'll go with this. We can change this afterward.

Reply:
How is that even working? That means we reconcile clusters twice 🤔